pallets/rc-migrator/src/proxy.rs

```diff
@@ -88,12 +88,35 @@ impl<T: Config> PalletMigration for ProxyProxiesMigrator<T> {
 
 			match Self::migrate_single(
 				acc.clone(),
-				(proxies.into_inner(), deposit),
+				(proxies.clone().into_inner(), deposit),
 				weight_counter,
 				&mut batch,
 			) {
 				Ok(proxy) => {
-					pallet_proxy::Proxies::<T>::remove(&acc);
+					// We keep proxy relations of pure accounts alive for free, otherwise gives the
+					// owner of the pure account a big headache with trying to control it through
+					// the remote proxy pallet (no UI for it) or similar.
+					if PureProxyCandidates::<T>::contains_key(&acc) {
+						// We dont remove it to keep it idempotent.
+
+						let mut free_proxies: BoundedVec<_, _> = proxies
+							.into_iter()
+							.filter(|p| T::PureProxyFreeVariants::contains(&p.proxy_type))
+							.collect::<Vec<_>>()
+							.defensive_truncate_into();
+						let deposit: BalanceOf<T> = Zero::zero();
+						log::debug!(target: LOG_TARGET, "Pure account {} gets {} proxies for free: {:?}", acc.to_ss58check(), free_proxies.len(), free_proxies);
+
+						if !free_proxies.is_empty() {
+							pallet_proxy::Proxies::<T>::insert(&acc, (free_proxies, deposit));
```

Comment:
does not this break idempotency of the proxy stage?

Reply:
If we run it twice then we just filter the Any proxies again, but it will again return the Any proxies.  
I just checked the output after adding a log and it is inserting the exact same stuff in the second idempotent run.